include/z64save.h

```diff
@@ -401,13 +401,54 @@ typedef enum SceneLayer {
 } SceneLayer;
 
 #define IS_CUTSCENE_LAYER (gSaveContext.sceneLayer >= SCENE_LAYER_CUTSCENE_FIRST)
+#define GET_CUTSCENE_LAYER(index) (SCENE_LAYER_CUTSCENE_FIRST + (index & 0xF))
 
 typedef enum LinkAge {
     /* 0 */ LINK_AGE_ADULT,
     /* 1 */ LINK_AGE_CHILD
 } LinkAge;
 
 
+// Values that indicate that no cutscene script should be played
+// (or that an "unscripted" cutscene is playing). The names "night"
+// and "day" are leftover from the original meaning of the cutscene
+// index (called "day_time"), and only map select uses these values
+// to indicate time of day (to set gSaveContext.save.nightFlag).
+#define CS_INDEX_NIGHT 0x0000
+#define CS_INDEX_DAY 0x8000
+
+// Values 0xFFF0-0xFFFF indicate that a cutscene script should be played.
+// If the value of `nextCutsceneIndex` is 0xFFF0-0xFFFF on scene load,
+// `Play_Init` will copy the value to `gSaveContext.cutsceneIndex`, load a
+// corresponding scene layer and start the scripted cutscene in the scene layer
+// (except for the value 0xFFFD, which is special-cased to do nothing in `Play_Init`).
+// It loads layer 4 for 0xFFF0, layer 5 for 0xFFF1, and so on.
+//
+// The cutsceneIndex could also be set to one of these values to start a
+// scripted cutscene immediately. In the vanilla game, this is used to play
+// the cutscene where the barrier in Ganon's Castle is dispelled (using index 0xFFFF)
+// and to preview cutscenes in debug mode (using index 0xFFFD).
```

Comment:
I don't like how poorly the function of 0xFFFD is being explained. Again, 0xFFFD is not scripted to "do nothing", it flushes the current scripted cutscene if something weird happens during the middle of the cutscene (e.g. the player falls out of bounds). It's the only appropriate value to use to avoid accidentally loading a layer unintentionally in such a scenario.

Reply:
Yeah we understand but do you have a suggestion on how to word it?

Reply:
thoughts?
```
// 0xFFFD also is used by the cutscene system to indicate a scripted cutscene has been triggered.
// This is why `Play_Init` ignores that value, so that if the cutscene is interrupted
// by a new play state load (e.g. if the player falls out of bounds), it does not then
// try to wrongly load a cutscene layer.
```

Reply:
Pushed this comment.